### PR TITLE
Specify different directory for `qq fit`

### LINKED ISSUE #773: Provide folder for `qq fit`

### PR DESCRIPTION
**Pull request by andrea-pasquale**
Closes #773 

- [x] Add tests

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Review by Edoardo-Pedicillo — APPROVED** [review]

File: src/qibocal/cli/fit.py [hunk 1]
```diff
@@ -1,26 +1,53 @@
...
 
-def fit(path, update):
+def fit(input_path, update, output_path, force):
     """Post-processing analysis
 
     Arguments:
 
-    - FOLDER: input folder.
-
+    - path: input folder.
```
Comment (GabrielePalazzo) [inline src/qibocal/cli/fit.py:25]:
```suggestion
    - input_path: input folder.
```

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import pathlib
+
+import pytest
+from click.testing import CliRunner
+
+from qibocal.cli._base import command
+
+DUMMY_ACTION = pathlib.Path(__file__).parent / "runcards/dummy_action.yml"
```
Comment (GabrielePalazzo) [inline tests/test_cli.py:8]:
```suggestion
DUMMY_ACTION = pathlib.Path(__file__).parent / "runcards" / "dummy_action.yml"
```

**Review by GabrielePalazzo — CHANGES_REQUESTED** [review]

File: doc/source/getting-started/interface.rst [hunk 3]
```diff
@@ -42,6 +42,11 @@ This command is used to perform a post-processing analysis on a given output fol
 
     qq fit <output_folder>
 
+If the ``<output-folder>`` already contains post-processing files they can be overwritten using ``-f``.
+It is also possible to generate a separated folder with the output folder generated by qibocal with
```
Comment (GabrielePalazzo) [inline doc/source/getting-started/interface.rst:46]: Did you mean "separate folder"? Separated is an adjective too, but it has a slightly different meaning.

File: src/qibocal/cli/fit.py [hunk 4]
```diff
@@ -1,26 +1,53 @@
...
+        if output_path.exists():
+            if force:
+                # overwrite output_path
+                log.warning(f"Deleting previous directory {output_path}.")
+                shutil.rmtree(os.path.join(os.getcwd(), output_path))
+                path = shutil.copytree(input_path, output_path)
+            else:
+                raise_error(RuntimeError, f"Directory {output_path} already exists.")
+        else:
+            path = shutil.copytree(input_path, output_path)
```
Comment (GabrielePalazzo) [inline src/qibocal/cli/fit.py:40]: This should do the same things with less lines of code:
```suggestion
        if output_path.exists():
            if force is False:
                raise_error(RuntimeError, f"Directory {output_path} already exists.")
            # overwrite output_path
            log.warning(f"Deleting previous directory {output_path}.")
            shutil.rmtree(pathlib.Path.cwd() / output_path)
        path = shutil.copytree(input_path, output_path)
```

**Review by GabrielePalazzo — APPROVED** [review]
Apart from https://github.com/qiboteam/qibocal/pull/774/files#r1544149133, it looks good to me.

**Comment by andrea-pasquale** [comment]
> Apart from https://github.com/qiboteam/qibocal/pull/774/files#r1544149133, it looks good to me.

Thanks @GabrielePalazzo. I saw the comment after asking you for the second review :)

**Comment by GabrielePalazzo** [comment]
> > Apart from https://github.com/qiboteam/qibocal/pull/774/files#r1544149133, it looks good to me.
> 
> Thanks @GabrielePalazzo. I saw the comment after asking you for the second review :)

I haven't checked if `pathlib` is imported.

**Pull request merged**